### DISCUSSION TITLE
[Feature Request] New Variable {{empty}}

### DISCUSSION THREAD

**Discussion by FelipeRearden** [Taitava/obsidian-shellcommands#142] [Ideas - on hold]
Hello @Taitava !!!!!

This idea came from using the new Modal Output with an `empty` Text field instead with something from {{selection}} or {{clipboard}}

So the new variable {{empty}} would be used with the Modal output  to open the Modal without any value. I was thinking about something like `echo {{empty}}` 

Let me know what you think :)

Have a great day!!!!!

**Comment by Taitava** [comment]
Hi Felipe,

would the idea be that `{{empty}}` would always give the same value: no value at all, just an empty text? Maybe I didn't understand this correctly. To me it sounds like this does not need any variable at all.

I need to think about this a bit.
 - Someting like `echo ""` with output directed to modal should be able to result in an (almost) empty text field in the modal. `echo` may output a newline character, but there might be some option to turn it off - maybe `echo -n ""`. Or another way: `echo "" > /dev/null` which dismisses output, resulting in completely empty output.
 - A current issue for now (during `0.10.0-beta.1`) is that the output modal does not open up if the output is completely empty. But if the output does contain even just whitespace (= a space, tab or newline), then the modal opens.

I think I'll change the modal to open up even with completely empty output in `stdout`. In a later phase there can be an option to prevent opening if output is empty.

**Reply by FelipeRearden** [reply]
> would the idea be that `{{empty}}` would always give the same value: no value at all, just an empty text? Maybe I didn't understand this correctly. To me it sounds like this does not need any variable at all.

That exactly this @Taitava just an empty text :)

One of my workflows will be open the modal empty, write some text and send somewhere else :)

> I think I'll change the modal to open up even with completely empty output in `stdout`. In a later phase there can be an option to prevent opening if output is empty.

Great idea @Taitava !!! I just thought that create a new variable would be easier than change the modal behavior :)

**Reply by Taitava** [reply]
> > would the idea be that `{{empty}}` would always give the same value: no value at all, just an empty text? Maybe I didn't understand this correctly. To me it sounds like this does not need any variable at all.
> 
> That exactly this @Taitava just an empty text :)

So you will not need a variable for this, just use `""` in a shell command. I'll move this discussion to _Ideas - on hold_, as if I understood correctly, there's nothing here I can actually develop. 🙂 

> 
> One of my workflows will be open the modal empty, write some text and send somewhere else :)

But that you will be able to accomplish with #17 / #37 (maybe even more easily than with the output modal). I'm currently working on it.

> 
> > I think I'll change the modal to open up even with completely empty output in `stdout`. In a later phase there can be an option to prevent opening if output is empty.
> 
> Great idea @Taitava !!! I just thought that create a new variable would be easier than change the modal behavior :)

I forgot to change the modal to open up even with completely empty output for `0.10.0`. So now it is like it is. But I'll create a separate feature idea discussion for an option that a user can toggle.
